unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCFriendsApiBridge.cs

```diff
@@ -240,7 +312,7 @@ public async UniTask<RendererCancelFriendRequestReply> CancelFriendRequest(Rende
         [PublicAPI]
         public async UniTask<ReceiveFriendRequestReply> ReceiveFriendRequest(ReceiveFriendRequestPayload request, RPCContext context, CancellationToken ct)
         {
-            await UniTask.SwitchToMainThread();
+            await UniTask.SwitchToMainThread(ct);
```

Comment:
can you provide an example?

Reply:
[Here](https://github.com/decentraland/unity-renderer/pull/4082/commits/b977a001c43a2eb861b44173a14b9a8b24bc43fa) is what I have in mind: thus, you clearly state that it is an action that should be invoked on the main thread but return the reply immediately.
Does it make sense, what do you think?

Reply:
in this case, i would have to switch to main thread twice, since im invoking two callbacks here. I dont feel its an improvement

Reply:
> i would have to switch to main thread twice, since im invoking two callbacks here

It is a valid point. Also in my piece, no cancellation token can participate.
I don't know at the moment how to solve it better. So let's drop my suggestion then.